### PR TITLE
Add unprivileged upgrade testing for Windows.

### LINKED ISSUE #4668: Uninstall on Windows unprivileged mode can fail to kill watchers
When Elastic Agent is installed in unprivileged mode on Windows and uninstall is performed it is possible that uninstall will fail to find/kill any running watchers.

```
[  ==] Successfully stopped service  [0s] 
                                                

[ ===] Stopping upgrade watcher (4284)  [0s] 
                                                

[ ===] Failed to find and stop watcher processes  [0s] 
                                                       

[ ===] Failed to uninstall current Elastic Agent  [0s] Error uninstalling. Printing logs
        Error: error installing package: failed to uninstall Agent at (C:\Program Files\Elastic): failed trying to kill any running watcher: failed to kill watcher process with pid 4284: DuplicateHandle: Access is denied.
        For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.14/fleet-troubleshooting.html
        
    logs_ingestion_test.go:90: 
        	Error Trace:	C:/Users/windows/agent/testing/integration/logs_ingestion_test.go:90
        	Error:      	Received unexpected error:
        	            	unable to enroll Elastic Agent: error running agent install command: exit status 1
        	Test:       	TestLogIngestionFleetManaged
--- FAIL: TestLogIngestionFleetManaged (11.20s)
```

This was identified from the PR that adds upgrade testing for Windows in unprivileged mode. https://github.com/elastic/elastic-agent/pull/4642

For confirmed bugs, please report:
- Version: 8.14+
- Operating System: Windows (unprivileged)
-

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Adds upgrade testing for unprivileged mode on Windows.

## Why is it important?

Upgrade testing for unprivileged mode on Windows was not enabled.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- [x] I have added an integration test or an E2E test

## Related issues

- Closes #4668

### CODE REVIEW

File: testing/integration/upgrade_fleet_test.go [hunk 1]
```diff
@@ -215,14 +215,19 @@ func testUpgradeFleetManagedElasticAgent(
...
+			t.Skipf("Starting version %s is less than 8.13-SNAPSHOT and doesn't support --unprivileged", startParsedVersion.String())
 		}
-		if endParsedVersion.Less(*upgradetest.Version_8_13_0) {
-			t.Skipf("Ending version %s is less than 8.13 and doesn't support --unprivileged", endParsedVersion.String())
+		if endParsedVersion.Less(*upgradetest.Version_8_13_0_SNAPSHOT) {
+			t.Skipf("Ending version %s is less than 8.13-SNAPSHOT and doesn't support --unprivileged", endParsedVersion.String())
 		}
 		if runtime.GOOS != define.Linux {
-			t.Skip("Unprivileged mode is currently only supported on Linux")
+			if startParsedVersion.Less(*upgradetest.Version_8_14_0_SNAPSHOT) {
```
Comment (cmacknz) [inline testing/integration/upgrade_fleet_test.go:225]:
```go
if runtime.GOOS != define.Linux {
```

Why not `if runtime.GOOS == define.Windows`? This block could have some confusing error messages if we start testing Macs. At the same time that's likely to be far enough in the future it won't hit this version constraint.

File: testing/integration/upgrade_standalone_test.go [hunk 2]
```diff
@@ -36,14 +36,13 @@ func TestStandaloneUpgrade(t *testing.T) {
...
-		unprivilegedAvailable := true
-		if runtime.GOOS != define.Linux {
-			// only available on Linux at the moment
-			unprivilegedAvailable = false
-		}
-		if unprivilegedAvailable && (startVersion.Less(*upgradetest.Version_8_13_0) || endVersion.Less(*upgradetest.Version_8_13_0)) {
-			// only available if both versions are 8.13+
-			unprivilegedAvailable = false
+		unprivilegedAvailable := false
+		if runtime.GOOS == define.Linux && !startVersion.Less(*upgradetest.Version_8_13_0_SNAPSHOT) && !endVersion.Less(*upgradetest.Version_8_13_0_SNAPSHOT) {
```
Comment (cmacknz) [inline testing/integration/upgrade_standalone_test.go:40]: Could be worth folding the `if` conditions here into a `supportsUnprivileged` helper so you don't have to repeat it multiple times.

**Review by cmacknz — APPROVED** [review]
LGTM besides a couple of nits.

File: testing/integration/upgrade_standalone_test.go (hunk 2 above)
Comment (blakerouse) [reply testing/integration/upgrade_standalone_test.go:40]: True, will add helper for this.

Comment (blakerouse) [reply testing/integration/upgrade_standalone_test.go]: Helper added.

Comment (cmacknz) [reply testing/integration/upgrade_standalone_test.go]: Thanks, looks tidier now.

**Comment by ycombinator** [comment]
Force merging as SonarQube is reporting 0% but new code added in the PR is not unit testable and integration tests have been updated.

**Pull request merged**